MediaSession.cpp

```diff
@@ -543,24 +539,29 @@ CDMi_RESULT MediaKeySession::Decrypt(
     }
     
     DRM_RESULT err = DRM_SUCCESS;
-    if (!initWithLast15) {
-/* PRv3.3 support */
-#ifdef PR_3_3
-      DRM_DWORD rgdwMappings[2];
-      if( f_pcbOpaqueClearContent == NULL || f_ppbOpaqueClearContent == NULL )
-      {
-          dr = DRM_E_INVALIDARG;
-          goto ErrorExit;
-      }
+    DRM_AES_COUNTER_MODE_CONTEXT ctrContext = { 0 };
+    DRM_DWORD rgdwMappings[2];
 
-      *f_pcbOpaqueClearContent = 0;
-      *f_ppbOpaqueClearContent = NULL;
+    if( f_pcbOpaqueClearContent == NULL || f_ppbOpaqueClearContent == NULL )
+    {
+        err = DRM_E_INVALIDARG;
+    }
 
-      ChkBOOL(m_eKeyState == KEY_READY, DRM_E_INVALIDARG);
-      ChkArg(f_pbIV != NULL && f_cbIV == sizeof(DRM_UINT64));
-#else
+    if(m_eKeyState != KEY_READY) err = DRM_E_INVALIDARG;
+    if(f_pbIV == NULL || f_cbIV != sizeof(DRM_UINT64)) err = DRM_E_INVALIDARG;
+
+    if (DRM_FAILED(err))
+    {
+        fprintf(stderr, "Error: Decrypt - Invalid argument\n");
+        return CDMi_S_FALSE;
+    }
```

Comment:
Please merge all of these with the above and return straight away without setting err and checking it separately.